src/lxc/conf.c

```diff
@@ -513,6 +513,9 @@ int lxc_storage_prepare(struct lxc_conf *conf)
 				 rootfs->path, rootfs->mount,
 				 rootfs->mnt_opts.raw_options ? rootfs->mnt_opts.raw_options : "(null)");
 
+	if (!rootfs->bdev_type)
+		rootfs->bdev_type = strdup(rootfs->storage->type);
```

Comment:
I don't understand. `lxc_rootfs_init()` explicitly allows for `rootfs->bdev_type` to be empty. If it's empty it means that it's supposed to be a directory.

Reply:
What's the config file that produces this issue?

Reply:
Ok, I pushed a fix on top of your commit and added your Co-Developed-by and Signed-off-by.